functions/New-DbaAgentJob.ps1

```diff
@@ -0,0 +1,476 @@
+Specifies when to delete the job.
+Allowed values 0, "Never", 1, "OnSuccess", 2, "OnFailure", 3, "Always"
+The text value van either be lowercase, uppercase or something in between as long as the text is correct.
+
+.NOTES 
+Original Author: Sander Stad (@sqlstad, sqlstad.nl)
+dbatools PowerShell module (https://dbatools.io, clemaire@gmail.com)
+Copyright (C) 2016 Chrissy LeMaire
+This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+You should have received a copy of the GNU General Public License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+.LINK
+https://dbatools.io/New-DbaAgentJob
+
+.EXAMPLE   
+New-DbaAgentJob -SqlServer 'sql1' -JobName 'Job1' -Description 'Just another job'
+Creates a job with the name "Job1" and a small description
+
+.EXAMPLE   
+New-DbaAgentJob -SqlServer 'sql1' -JobName 'Job1' -Disabled
+Creates the job but sets it to disabled
+
+.EXAMPLE
+New-DbaAgentJob -SqlServer 'sql1' -JobName 'Job1' -EventLogLevel 'OnSuccess'
+Creates the job and sets the notification to write to the Windows Application event log on success
+
+.EXAMPLE
+New-DbaAgentJob -SqlServer 'SSTAD-PC' -JobName 'Job1' -EmailLevel 'OnFailure' -EmailOperatorName 'dba'
+Creates the job and sets the notification to send an e-mail to the e-mail operator
+
+.EXAMPLE   
+New-DbaAgentJob -SqlServer 'sql1' -JobName 'Job1' -Description 'Just another job' -Whatif
+Doesn't create the job but shows what would happen.
+
+.EXAMPLE   
+"sql1", "sql2", "sql3" | New-DbaAgentJob -JobName 'Job1'
+Creates a job with the name "Job1" on multiple servers
+#>
+
+    [CmdletBinding(DefaultParameterSetName = "Default", SupportsShouldProcess = $true)]
+
+    param(
+        [parameter(Mandatory = $true, ValueFromPipeline=$true)]
+		[object[]]$SqlServer,
+        [Parameter(Mandatory = $false)]
+        [System.Management.Automation.PSCredential]$SqlCredential,
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [string]$JobName,
+        [Parameter(Mandatory = $false)]
+        [switch]$Disabled,
+        [Parameter(Mandatory = $false)]
+        [string]$Description,
+        [Parameter(Mandatory = $false)]
+        [int]$StartStepId,
+        [Parameter(Mandatory = $false)]
+        [string]$CategoryName,
+        [Parameter(Mandatory = $false)]
+        [int]$CategoryId,
+        [Parameter(Mandatory = $false)]
+        [string]$OwnerLoginName,
+        [Parameter(Mandatory = $false)]
+        [ValidateSet(0, "Never", 1, "OnSuccess", 2, "OnFailure", 3, "Always")]
+        [object]$EventLogLevel = '',
```

Comment:
The reason I did that is because I'm checking if the value submitted is text and than do a switch to get the right integer value. I could probably set it to 0 which is the default for sql server